datanode/service/market.go

```diff
@@ -55,8 +55,8 @@ func (m *Markets) Upsert(ctx context.Context, market *entities.Market) error {
 		return err
 	}
 	m.cacheLock.Lock()
-	if market.State == entities.MarketStateSettled || market.State == entities.MarketStateRejected {
-		// a settled or rejected market can be safely removed from this map.
+	if market.State == entities.MarketStateSettled || market.State == entities.MarketStateRejected || market.State == entities.MarketStateCancelled {
```

Comment:
@ze97286 This seems to be specific to spot markets, I can't really see why cancelled shouldn't be included here, so I added it, but would be good to get confirmation.

Reply:
I think that's fine